src/components/PipelineRun/PipelineRunDetailsView/visualization/utils/pipelinerun-graph-utils.ts

```diff
@@ -354,11 +471,37 @@ const getGraphDataModel = (
         });
       }
 
+      // Expand matrix task dependencies
+      const expandedRunAfterTasks = expandMatrixDependencies(runAfterTasks);
+
+      // For matrix tasks, use a display name that includes platform info
+      const matrixTask = task as MatrixPipelineTaskWithStatus;
+      const displayName = matrixTask.matrixPlatform
+        ? `${matrixTask.originalName} (${matrixTask.matrixPlatform})`
+        : task.name;
+
+      // For matrix tasks, find the specific TaskRun for this platform
+      const findTaskRun = (): TaskRunKind | undefined => {
+        if (matrixTask.matrixPlatform) {
+          // Matrix task - find TaskRun with matching platform label
+          const platformLabel = matrixTask.matrixPlatform.replace(/\//g, '-');
+          return taskRuns.find(
+            (tr) =>
+              tr.metadata.labels[TektonResourceLabel.pipelineTask] === matrixTask.originalName &&
+              tr.metadata.labels[TaskRunLabel.TARGET_PLATFORM] === platformLabel,
+          );
+        }
+        // Regular task - find by task name
+        return taskRuns.find(
+          (tr) => tr.metadata.labels[TektonResourceLabel.pipelineTask] === task.name,
+        );
+      };
```

Comment:
This doesn't have to be a method. just an if else condition would be enough to find the task run.